### PR TITLE
[9.4] Use old rolling upgrade strategy when protocol < 2.5

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.jboss.org/browse/ISPN-10165

### CODE REVIEW

**Comment by tristantarrant** [comment]
@andyuk1986 @karesti this reintroduces the old rolling upgrade strategy

File: server/integration/testsuite/src/test/java/org/infinispan/server/test/rollingupgrades/HotRodRollingUpgradesDistIT.java [hunk 1]
```diff
@@ -90,12 +92,21 @@ public void testHotRodRollingUpgradesDiffVersionsDist() throws Exception {
             assertEquals("Can't access entries stored in source node (target's RemoteCacheStore).", "value1", c2.get("key1"));
             assertEquals("Can't access entries stored in source node (target's RemoteCacheStore).", "valuex1", c2.get("keyx1"));
 
-            provider1 = new MBeanServerConnectionProvider(s1.server.getHotrodEndpoint().getInetAddress().getHostName(),
+            MBeanServerConnectionProvider provider1 = new MBeanServerConnectionProvider(s1.server.getHotrodEndpoint().getInetAddress().getHostName(),
```
Comment (galderz) [inline server/integration/testsuite/src/test/java/org/infinispan/server/test/rollingupgrades/HotRodRollingUpgradesDistIT.java]: This and the one after should be closed after use?

Comment (tristantarrant) [reply server/integration/testsuite/src/test/java/org/infinispan/server/test/rollingupgrades/HotRodRollingUpgradesDistIT.java]: No, there is no close() method on the underlying connection.

**Comment by tristantarrant** [comment]
I have issued a PR for infinispan-arquillian which will need a release and then I'll issue a separate PR to wrap all MBeanServerConnectionProvider calls in a try-with-arm block

**Comment by karesti** [comment]
integarted, thanks @tristantarrant !

**Pull request merged**